src/Service/UserTransformer.php

```diff
@@ -40,39 +40,54 @@ public function __construct(
         $this->techRepo = $techRepo;
         $this->encoder = $encoder;
     }
-
     /**
      * @param UserDTO $dto
      * @return User
      * @throws EntityNotFound
      */
-    public function transform(
+    public function registerTransform(
         UserDTO $dto
     ): User {
-        if ($dto->id !== null) {
-            $entity = $this->userRepo->find($dto->id);
-            if (!$entity) {
-                $entityNotFound = new EntityNotFound(
-                    User::class,
-                    $dto->id,
-                    'No user found.'
-                );
-                throw $entityNotFound;
-            }
-            foreach ($entity->getTechnologies() as $techToRemove) {
-                $entity->removeTechnology($techToRemove);
-            }
-        } else {
-            $entity = new User();
-        }
+        $entity = new User();
         $entity->setName($dto->name);
         $entity->setPassword($this->encoder->encodePassword($entity, $dto->password));
         $entity->setSurname($dto->surname);
         $entity->setUsername($dto->username);
         $entity->setCreatedAt($dto->createdAt);
         $entity->setUpdatedAt($dto->updatedAt);
         $entity->setEmail($dto->email);
+        return $entity;
+    }
 
+    /**
+     * @param UserDTO $dto
+     * @return User
+     * @throws EntityNotFound
+     */
+    public function editTransform(
+        UserDTO $dto,
+        $id
```

Comment:
Send the User object, not the id. You already searched the user in the database, there's no reason to do it twice.